engine/storage/image/src/org/apache/cloudstack/storage/image/TemplateServiceImpl.java

```diff
@@ -324,6 +333,17 @@ public void handleTemplateSync(DataStore store) {
                         }
                     }
 
+                    for (Iterator<VMTemplateVO> iter = allTemplates.listIterator(); iter.hasNext();) {
+                        VMTemplateVO child_template = iter.next();
+                        if (child_template.getParentTemplateId() != null) {
+                            String uniqueName = child_template.getUniqueName();
+                            if (templateInfos.containsKey(uniqueName)) {
+                                templateInfos.remove(uniqueName);
+                            }
+                            iter.remove();
+                        }
+                    }
+
                     toBeDownloaded.addAll(allTemplates);
```

Comment:
Won't this `allTemplates` be empty after the previous code?

Reply:
As per my understanding, only parent templates (parent_template_id = null) will be on allTemplates after previous code

Reply:
Ah yes. My bad...
I got lost with the IFs...